lib/const.rb

```diff
@@ -1,113 +1,116 @@
+# frozen_string_literal: true
+
 # Defines common constants used in different parts of crew
 
 CREW_VERSION = '1.23.16'
 
 ARCH_ACTUAL = `uname -m`.chomp
 # This helps with virtualized builds on aarch64 machines
 # which report armv8l when linux32 is run.
-ARCH = ( ARCH_ACTUAL == 'armv8l' ) ? 'armv7l' : ARCH_ACTUAL
+ARCH = ARCH_ACTUAL == 'armv8l' ? 'armv7l' : ARCH_ACTUAL
 
 # Allow for edge case of i686 install on a x86_64 host before linux32 is
 # downloaded, e.g. in a docker container.
-ARCH_LIB = ( ARCH == 'x86_64' and Dir.exist?('/lib64') ) ? 'lib64' : 'lib'
+ARCH_LIB = ARCH == 'x86_64' && Dir.exist?('/lib64') ? 'lib64' : 'lib'
 
 # Glibc version can be found from the output of libc.so.6
-LIBC_VERSION = %x[/#{ARCH_LIB}/libc.so.6][/Gentoo ([^-]+)/, 1]
+LIBC_VERSION = `/#{ARCH_LIB}/libc.so.6`[/Gentoo ([^-]+)/, 1]
 
-unless ENV['CREW_PREFIX'].to_s.empty? or ENV['CREW_PREFIX'] == '/usr/local'
-  CREW_BUILD_FROM_SOURCE = 1
-  CREW_PREFIX = ENV['CREW_PREFIX']
-  HOME = CREW_PREFIX + ENV['HOME']
-else
-  CREW_BUILD_FROM_SOURCE = ENV['CREW_BUILD_FROM_SOURCE']
+if ENV['CREW_PREFIX'].to_s.empty? || ENV['CREW_PREFIX'] == '/usr/local'
+  CREW_BUILD_FROM_SOURCE = ENV.fetch('CREW_BUILD_FROM_SOURCE', nil)
   CREW_PREFIX = '/usr/local'
-  HOME = ENV['HOME']
+  HOME = Dir.home
+else
+  CREW_BUILD_FROM_SOURCE = 1
+  CREW_PREFIX = ENV.fetch('CREW_PREFIX', nil)
+  HOME = "#{CREW_PREFIX}#{Dir.home}"
 end
 
-CREW_LIB_PREFIX = CREW_PREFIX + '/' + ARCH_LIB
-CREW_MAN_PREFIX = CREW_PREFIX + '/share/man'
-CREW_LIB_PATH = CREW_PREFIX + '/lib/crew/'
-CREW_PACKAGES_PATH = CREW_LIB_PATH + 'packages/'
-CREW_CONFIG_PATH = CREW_PREFIX + '/etc/crew/'
-CREW_META_PATH = CREW_CONFIG_PATH + 'meta/'
-CREW_BREW_DIR = CREW_PREFIX + '/tmp/crew/'
-CREW_DEST_DIR = CREW_BREW_DIR + 'dest'
-CREW_DEST_PREFIX = CREW_DEST_DIR + CREW_PREFIX
-CREW_DEST_LIB_PREFIX = CREW_DEST_DIR + CREW_LIB_PREFIX
-CREW_DEST_MAN_PREFIX = CREW_DEST_DIR + CREW_MAN_PREFIX
+CREW_LIB_PREFIX = "#{CREW_PREFIX}/#{ARCH_LIB}"
+CREW_MAN_PREFIX = "#{CREW_PREFIX}/share/man"
+CREW_LIB_PATH = "#{CREW_PREFIX}/lib/crew"
+CREW_PACKAGES_PATH = "#{CREW_LIB_PATH}/packages"
+CREW_CONFIG_PATH = "#{CREW_PREFIX}/etc/crew"
+CREW_META_PATH = "#{CREW_CONFIG_PATH}meta"
+CREW_BREW_DIR = "#{CREW_PREFIX}/tmp/crew"
+CREW_DEST_DIR = "#{CREW_BREW_DIR}/dest"
+CREW_DEST_PREFIX = "#{CREW_DEST_DIR}/#{CREW_PREFIX}"
+CREW_DEST_LIB_PREFIX = "#{CREW_DEST_DIR}/#{CREW_LIB_PREFIX}"
+CREW_DEST_MAN_PREFIX = "#{CREW_DEST_DIR}/#{CREW_MAN_PREFIX}"
```

Comment:
This is a good consideration since there could be spaces in the directories.  I would never recommend using spaces in a directory but it is a possibility.